cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowProgramInfo.java

```diff
@@ -53,23 +54,25 @@ public static WorkflowProgramInfo create(Arguments arguments) {
     String workflowRunId = arguments.getOption(ProgramOptionConstants.WORKFLOW_RUN_ID);
     String programNameInWorkflow = arguments.getOption(ProgramOptionConstants.PROGRAM_NAME_IN_WORKFLOW);
     String workflowToken = arguments.getOption(ProgramOptionConstants.WORKFLOW_TOKEN);
-
+    boolean disableFieldLineageEmitFromNodes = Boolean.parseBoolean(ProgramOptionConstants.DISABLE_FIELD_LINEAGE_EMIT);
```

Comment:
that would be always false? Because that String is "disableFieldLineageEmit". Don't you want to parse arguments.getOption(ProgramOptionConstants.DISABLE_FIELD_LINEAGE_EMIT)? and only if it is not null?